ProcessLib/ComponentTransport/ComponentTransportFEM.h

```diff
@@ -98,19 +110,19 @@ class LocalAssemblerData : public ComponentTransportLocalAssemblerInterface
         bool is_axially_symmetric,
         unsigned const integration_order,
         ComponentTransportProcessData const& process_data,
-        std::vector<std::reference_wrapper<ProcessVariable>> const&
-            per_process_variables)
+        std::vector<std::vector<std::reference_wrapper<ProcessVariable>>> const&
+            process_variables)
         : _element(element),
           _process_data(process_data),
           _integration_method(integration_order),
-          _per_process_variables(per_process_variables)
+          _process_variables(process_variables)
     {
         // This assertion is valid only if all nodal d.o.f. use the same shape
         // matrices.
-        // _per_process_variables.size() can represent number of nodal DOFs
+        // process_variables[0].size() can represent number of nodal DOFs
         // regardless of what scheme is adopted.
         assert(local_matrix_size ==
-               ShapeFunction::NPOINTS * _per_process_variables.size());
+               ShapeFunction::NPOINTS * _process_variables[0].size());
```

Comment:
... and this `[0]` is some process id.
It would be better, if the `[0]` accesses would be replaced with corresponding process or component ids.

Reply:
It is hard to resolve...

Reply:
Why not give the correct process variable to the `LocalAssemblerData` constructor instead of a vector of vector of `ProcessVariable`?

Reply:
For staggered coupling scheme, the information of process variables is no longer queued in a non-nesting vector container.

Reply:
You could pick `hydraulic_process_variable` and `component_process_variables` from the `process_variables` vector.  In the monolithic implementation the process_variables could be the same. What do you think?

Reply:
Following your suggestion, I have made changes accordingly.